### PR TITLE
Initialize the grouping sets when there is a group by all to enable filter pushdown

### PR DESCRIPTION
**Pull request by Tmonster**
Fixes https://github.com/duckdblabs/duckdb-internal/issues/3316

The `GROUP BY ALL` doesn't add grouping sets to the logical aggregate like a regular `GROUP BY` does. 
The grouping sets indexes come from how many grouping expressions there are, and not how many group_indexes there are. This is because in the presence of a HAVING clause, the Having Binder will add another group by expression that is not picked up when the `groupby_all_indexes` are calculated.

### CODE REVIEW

File: src/planner/binder/query_node/bind_select_node.cpp [hunk 1]
```diff
@@ -616,13 +616,28 @@ unique_ptr<BoundQueryNode> Binder::BindSelectNode(SelectNode &statement, unique_
...
+	set<idx_t> group_by_all_indexes_set;
+	if (!group_by_all_indexes.empty() && !statement.having) {
+		for (idx_t i = 0; i < group_by_all_indexes.size(); i++) {
+			group_by_all_indexes_set.insert(i);
+		}
+		D_ASSERT(result->groups.grouping_sets.empty());
+		result->groups.grouping_sets.push_back(group_by_all_indexes_set);
+		// rebind the having in this case
+		if (statement.having) {
+			HavingBinder having_binder(*this, context, *result, info, statement.aggregate_handling);
```
Comment (Mytherin) [inline src/planner/binder/query_node/bind_select_node.cpp:636]: I'm a bit unsure about this part - binding can be destructive, so rebinding the same expression again is not necessarily safe.

Could you expand a bit on why this is necessary?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - one comment

**Comment by Tmonster** [comment]
The group by all with having was failing and I thought maybe the having statement inspects the grouping sets and the group expressions, but it turns out that's not the case. Also, re-binding the having statement doesn't even happen because of [this line](https://github.com/duckdb/duckdb/pull/14660/files#diff-537b9045ad0cab2ecaddf9ccc3cd95e439fb88c6642af01dfd4d3653413605ccR628).

I've been looking into it more and I think there are some weird cases around group by all and having. Going to see if I can fix them here.

File: test/sql/aggregate/group/group_by_all_having.test [hunk 2]
```diff
@@ -3,18 +3,20 @@
...
 statement ok
-PRAGMA enable_verification
+PRAGMA disable_optimizer;
 
-query I
-SELECT * FROM (SELECT 1) t0(c0) GROUP BY c0 HAVING c0>0
-----
-1
+# query I
+# SELECT * FROM (SELECT 1) t0(c0) GROUP BY c0 HAVING c0>0
```
Comment (Mytherin) [inline test/sql/aggregate/group/group_by_all_having.test:9]: Could we revert these test skips?

File: src/planner/binder/query_node/bind_select_node.cpp [hunk 3]
```diff
@@ -616,13 +616,24 @@ unique_ptr<BoundQueryNode> Binder::BindSelectNode(SelectNode &statement, unique_
...
 		auto &expr = result->select_list[group_by_all_index];
 		auto group_ref = make_uniq<BoundColumnRefExpression>(
 		    expr->return_type, ColumnBinding(result->group_index, result->groups.group_expressions.size()));
 		result->groups.group_expressions.push_back(std::move(expr));
 		expr = std::move(group_ref);
 	}
+	set<idx_t> group_by_all_indexes_set;
+	if (!group_by_all_indexes.empty()) {
+		idx_t num_set_indexes = result->groups.group_expressions.size();
+		;
```
Comment (Mytherin) [inline src/planner/binder/query_node/bind_select_node.cpp:630]: nit

File: test/sql/aggregate/group/group_by_all_having.test [hunk 4]
```diff
@@ -3,7 +3,7 @@
 # group: [group]
 
 statement ok
-PRAGMA enable_verification
+PRAGMA disable_optimizer;
```
Comment (Mytherin) [inline test/sql/aggregate/group/group_by_all_having.test:6]: Possibly accidentally left in?

**Pull request merged**